src/guide/frontend/android.md

```diff
@@ -0,0 +1,340 @@
+```
+
+:::
+
+Put this SHA256 along with your root android package name in your backend to generate `assetlinks.json` for your app at `https://yourexample.com/.well-known/assetlinks.json`.
+If you are using `example-nodejs-backend`. then just put these 2 values in your `.env` file.
+
+### Host ~/.well-known/assetlinks.json
+
+You will need store the following file at `https://<your-domain>/.well-known/assetlinks.json`. To generate the SHA256 hash, read the links below the snippet.
+
+You may also have to change the 'target::namespace' and 'target::package_name' properties to match your application's.
+
+```json
+[
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "web"
+    }
+  },
+  {
+    "relation": [
+      "delegate_permission/common.handle_all_urls",
+      "delegate_permission/common.get_login_creds"
+    ],
+    "target": {
+      "namespace": "android_app",
+      "package_name": "com.example.myapplication",
+      "sha256_cert_fingerprints": [
+        "3C:E2:29:94:E2:DE:1E:EB:E5:F9:70:10:72:41:F4:0F:06:38:61:BD:72:76:79:CA:72:68:67:FA:38:9B:65:B9"
+      ]
+    }
+  }
+]
+```
+
+## Using the PasswordlessClient
+
+::: tabs
+
+@tab With Dagger Hilt
+
+You can either set the `ActivityContext` and `CoroutineScope` by injecting it with Dagger Hilt as follows:
+
+```kotlin
+@Module
+@InstallIn(ActivityComponent::class)
+class PasswordlessModule {
+    @Provides
+    fun provideLifecycleCoroutineScope(activity: Activity): LifecycleCoroutineScope =
+        (activity as AppCompatActivity).lifecycleScope
+
+    @Provides
+    @ActivityScoped
+    fun providePasswordlessClient(
+        @ActivityContext activity: Context, scope: LifecycleCoroutineScope): PasswordlessClient {
+        val options = PasswordlessOptions(
+            DemoPasswordlessOptions.API_KEY,
+            DemoPasswordlessOptions.RP_ID,
+            DemoPasswordlessOptions.ORIGIN,
+            DemoPasswordlessOptions.API_URL
+        )
+
+        return PasswordlessClient(options, activity, scope)
+    }
+}
+```
+
+@tab Without Dagger Hilt
+
+Or you can set the Context of PasswordlessClient manually: Ensure the context is set to the current `Activity`.
+
+```kotlin
+/** Context needs to be set according to current activity
+ * If there are different activity handling register and signin,
+ * then call this on every activity
+*/
+_passwordless.setContext(this)
+```
+
+Set the coroutine scope, passing lifecycleScope of the current fragment, only necessary if you again do not use Dagger Hilt.
+
+```kotlin
+_passwordless.setCoroutineScope(lifecycleScope)
+```
+
+:::
+
+## Registration
+
+1. **UI**:
+   - (Optional) Take the alias as input from the user if it's a non-discoverable credential.
+   - (Required) Specify the username for the credential.
```

Comment:
Would always recommend putting required steps above optional ones, unless the order they're done in matters.